[DOCS] Adds ml-cpp PRs to release notes (#52158)
elastic/elasticsearch: docs/reference/release-notes/7.6.asciidoc

* Improve model_memory_limit UX for data frame analytics jobs {pull}44699[#44699]
* Improve performance of boosted tree training for both classification and regression {ml-pull}775[#775]
* Reduce the peak memory used by boosted tree training and fix an overcounting bug
estimating maximum memory usage {ml-pull}781[#781]
* Stratified fractional cross validation for regression {ml-pull}784[#784]
* Added `geo_point` supported output for `lat_long` function records {ml-pull}809[#809], {pull}47a050[#47050]
* Use a random bag of the data to compute the loss function derivatives for each 
new tree which is trained for both regression and classification {ml-pull}811[#811]
* Emit `prediction_probability` field alongside prediction field in ml results {ml-pull}818[#818]
* Reduce memory usage of {ml} native processes on Windows {ml-pull}844[#844]
* Reduce runtime of classification and regression {ml-pull}863[#863]
* Stop early training a classification and regression forest when the validation 
error is no longer decreasing {ml-pull}875[#875]
* Emit `prediction_field_name` in ml results using the type provided as
`prediction_field_type` parameter {ml-pull}877[#877]

> * Improve performance updating quantile estimates {ml-pull}881[#881]

Reviewer: ```suggestion
* Improve performance of classification and regression {ml-pull}881[#881]
```

* Slash missed in indices.put_mapping url {pull}49468[#49468]

Machine Learning::
* Fix 2 digit year regex in find_file_structure {pull}51469[#51469]
* Validate classification dependent_variable cardinality is at lea… {pull}51232[#51232]
* Do not copy mapping from dependent variable to prediction field in regression analysis {pull}51227[#51227]
* Handle nested and aliased fields correctly when copying mapping. {pull}50918[#50918] (issue: {issue}50787[#50787])
* Fix off-by-one error in ml_classic tokenizer end offset {pull}50655[#50655]
* Improve uniqueness of result document IDs {pull}50644[#50644] (issue: {issue}50613[#50613])
* Synchronize processInStream.close() call {pull}50581[#50581] (issue: {issue}49680[#49680])

> * Fix accuracy metric {pull}50310[#50310] (issue: {issue}48759[#48759])

Reviewer: ```suggestion
* Fix accuracy metric in multi-class confusion matrix {pull}50310[#50310] (issue: {issue}48759[#48759])
```
